components/script/dom/htmlelement.rs

```diff
@@ -783,6 +786,49 @@ impl HTMLElement {
 
         None
     }
+
+    // https://html.spec.whatwg.org/multipage/#the-summary-element:activation-behaviour
+    pub fn summary_activation_behavior(&self) {
+        // Step 1
+        if !self.is_summary_for_its_parent_details() {
+            return;
+        }
+
+        // Step 2
+        let parent_details = self.upcast::<Node>().GetParentNode().unwrap();
+
+        // Step 3
+        parent_details
+            .downcast::<HTMLDetailsElement>()
+            .unwrap()
+            .toggle();
+    }
+
+    // https://html.spec.whatwg.org/multipage/#summary-for-its-parent-details
+    fn is_summary_for_its_parent_details(&self) -> bool {
+        // Step 1
+        let summary_node = self.upcast::<Node>();
+        if !summary_node.has_parent() {
+            return false;
+        }
+
+        // Step 2
+        let parent = &summary_node.GetParentNode().unwrap();
+
+        // Step 3
+        if parent.downcast::<Element>().unwrap().local_name() != &local_name!("details") {
+            return false;
+        }
+
+        // Step 4 & 5
+        let first_summary_element = parent
+            .child_elements()
+            .find(|el| el.local_name() == &local_name!("summary"));
+        match first_summary_element {
+            Some(first_summary) => first_summary == DomRoot::from_ref(self.upcast::<Element>()),
```

Comment:
I believe this could be `&*first_summary == self.upcast::<Element>())` instead.

Reply:
This works, even i dont yet understand what "&*first_summary" means.